File: Front/ProEventos-App/src/app/models/Palestrante.ts
import { Evento } from "./Evento";
import { RedeSocial } from "./rede-social";

export interface Palestrante {
  id: number;
  nome: string;
  minicurriculo: number;
  imagemURL: string;
  telefone: string;
  email: string;
  redeSociais: RedeSocial[];
  palestranteEventos: Evento[];
}

File: Back/src/ProEventos.Domain/Palestrante.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ProEventos.Domain
{
    public class Palestrante
    {
        [Key]
        public int Id { get; set; }
        public string Nome { get; set; }
        public int Minicurriculo { get; set; }
        public string ImagemURL { get; set; }
        public string Telefone { get; set; }
        public string Email { get; set; }
        public IEnumerable<RedeSocial> RedeSociais { get; set; }
        public IEnumerable<PalestranteEvento> PalestranteEventos { get; set; }
    }
}